cmd/namespace.go

```diff
@@ -190,8 +188,25 @@ func namespaceAssociate(ctx *cli.Context) error {
 		return err
 	}
 
+	projResource, err := Lookup(c, ctx.Args().Get(1), "project")
+	if nil != err {
+		return err
+	}
+
+	proj, err := getProjectByID(c, projResource.ID)
+	if nil != err {
+		return err
+	}
+
+	if _, ok := namespace.Actions["move"]; ok {
+		move := &clusterClient.NamespaceMove{
+			ProjectID: proj.ID,
+		}
+		return c.ClusterClient.Namespace.ActionMove(namespace, move)
+	}
+
 	update := make(map[string]string)
-	update["projectId"] = ctx.Args().Get(1)
+	update["projectId"] = proj.ID
```

Comment:
Don't have to update projectID. Move action should update it. And also projectID is non-updatable.

Reply:
This happens on versions before your change.

Reply:
@dramich there is a scenario where the new version of rancher is used AND the move action is not present.
THat is actually the primary reason for making it an action: so that the UI can pre-emptively tell if the namesapce can be moved.

So, I like that you added the backwards compatibility, but the logic might have to get more complicated.

I don;t know the solutoin, but here's my rough idea on how it should behave:
1. if move action is available, call it.
2. for backwards compat, if it is not available, you can try update, but update, I believe, will silently drop the update and return. So, when update returns, if the project ID remains the same, tell the user that the ns may not be moved

Reply:
Fixed by checking the annotations